src/clj/rems/workflow/dynamic.clj

```diff
@@ -658,12 +715,28 @@
         (is (not (:success result)))
         (is (= [{:type :forbidden}] (:errors result)))))))
 
+(defmacro assert-ex
+  "Like assert but throw the result with ex-info and not as string. "
+  ([x message]
+   `(when-not ~x
+      (throw (ex-info (str "Assert failed: " ~message "\n" (pr-str '~x))
+                      (merge ~message {:expression '~x}))))))
+
+
+(defmacro catch-ex
+  "Like `catch` but automatically unwraps possible exception to the result."
+  [& body]
```

Comment:
it's not really like `catch`, it's like `try`, right?

Reply:
Well it catches the exception so you don't have to. Maybe `try-catch-ex`?